internal/pkg/v2/infrastructure/redis/client.go

```diff
@@ -58,3 +58,20 @@ func (c *Client) AddEvent(e model.Event) (model.Event, errors.EdgeX) {
 
 	return addEvent(conn, e)
 }
+
+// Add a new device profle
+func (c *Client) AddDeviceProfile(dp model.DeviceProfile) (model.DeviceProfile, errors.EdgeX) {
+	conn := c.Pool.Get()
+	defer conn.Close()
+
+	if dp.Id != "" {
+		_, err := uuid.Parse(dp.Id)
+		if err != nil {
+			return model.DeviceProfile{}, errors.NewCommonEdgeX(errors.KindInvalidId, "uuid parsing failed", err)
```

Comment:
Change message to `ID failed UUID parsing`

Reply:
Modified